#! /bin/sh -e

# DP: GNU/k*BSD support
# Author: Aurelien Jarno <aurel32@debian.org>
# Status: Submitted: http://gcc.gnu.org/ml/gcc-patches/2007-04/msg01962.html

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
	cd ${dir}gcc && autoconf2.59
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
	rm -f ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

--- src/gcc/configure.ac.orig	2007-04-28 14:51:34.000000000 +0200
+++ src/gcc/configure.ac	2007-04-28 14:52:13.000000000 +0200
@@ -3093,7 +3093,7 @@
 
 # Test for stack protector support in target C library.
 case "$target" in
-  *-*-linux*)
+  *-*-gnu*|*-*-*-gnu*)
     AC_CACHE_CHECK(__stack_chk_fail in target GNU C library,
       gcc_cv_libc_provides_ssp,
       [gcc_cv_libc_provides_ssp=no
